File: 351003/Kisel/Apache Kafka/DiscussionService/Models/Dtos/PostMessage.cs
namespace DiscussionService.Models.Dtos;

public class PostMessage
{
    public string Action { get; set; } = string.Empty;
    public int Id { get; set; }
    public int NewsId { get; set; }
    public string Content { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
}

File: 351004/Nepesov/Lab4/untitled/src/main/java/org/example/dto/PostMessage.java
package org.example.dto;

import lombok.*;
import org.example.model.PostState;

@Data
@NoArgsConstructor
@AllArgsConstructor // Этот конструктор теперь будет принимать PostState
public class PostMessage {
    private Long id;
    private Long newsId;
    private String content;
    private PostState state; // Проверь, чтобы здесь не было String
}